Source/messaging/MessageClient.cpp

```diff
@@ -159,9 +160,11 @@ namespace Messaging {
      *
      * @param function function to be called on each of the messages in the buffer
      */
-    void MessageClient::PopMessagesAndCall(std::function<void(const Core::Messaging::IStore::Information& info, const Core::ProxyType<Core::Messaging::IEvent>& message)> function)
+    void MessageClient::PopMessagesAndCall(std::function<void(const Core::Messaging::Metadata& metadata, const Core::ProxyType<Core::Messaging::IEvent>& message)> function)
     {
-        Core::Messaging::IStore::Information information;
+        Core::Messaging::Metadata metadata;
+        Core::Messaging::IStore::Logging log;
+        Core::Messaging::IStore::Tracing trace;
```

Comment:
Initialize variables only when needed.

Reply:
Is it more efficient to initialize these variables multiple times inside the loop, but only when necessary, rather than just doing it once at the start?